code/modules/admin/admin_fax_panel.dm

```diff
@@ -0,0 +1,134 @@
+/client/proc/fax_panel()
+	set category = "Admin.Events"
+	set name = "Fax Panel"
+
+	admin_fax_panel()
+
+/client/proc/admin_fax_panel()
+	if(!check_rights(R_ADMIN))
+		return
+
+	var/datum/fax_panel_interface/ui = new(usr)
+	ui.ui_interact(usr)
+
+/// Panel
+/datum/fax_panel_interface
+	/// All faxes in game list
+	var/available_faxes = list()
+	/// List with available stamps
+	var/stamp_list = list() // I need to understand, how to remove chameleon stamps...
+
+	/// Paper which admin edit and send.
+	var/obj/item/paper/fax_paper = new /obj/item/paper(null)
+
+	/// Default name of fax. Used when field with fax name not edited.
+	var/sending_fax_name = "Secret"
+	/// Default name of paper. paper - bluh-bluh. Used when field with paper name not edited.
+	var/default_paper_name = "Standart Report"
+
+/datum/fax_panel_interface/New()
+	//Get all faxes, and save them to our list.
+	for(var/obj/machinery/fax/fax in GLOB.machines)
+		available_faxes += fax
+	//Get all stamps
+	for(var/stamp in subtypesof(/obj/item/stamp))
+		var/obj/item/stamp/real_stamp = stamp 
+		if(length(initial(real_stamp.actions)) == 0)// try to remove chameleon, dont work.
+			stamp_list += list(list(initial(real_stamp.name), initial(real_stamp.icon_state)))
+	//Give our paper special status, to read everywhere.
+	fax_paper.request_state = TRUE
+
+//Maybe.. useless proc, but find fax like object what we need.
+/datum/fax_panel_interface/proc/get_fax_by_name(name)
+	if(!length(available_faxes))
+		return
+
+	for(var/obj/machinery/fax/potential_fax in available_faxes)
+		if(potential_fax.fax_name == name)
+			return potential_fax
+	return
```

Comment:
Make these 2 blank returns explicitly `return null` instead, to indicate we always return a value.